javascript/hangman.js

```diff
@@ -2,37 +2,62 @@ class Hangman {
   constructor(words) {
     this.words = words;
     // ... your code goes here
+    this.secretWord = words[Math.floor(Math.random() * words.length)];
+    this.letters = [];
+    this.guessedLetters = "";
+    this.errorsLeft = 10;
   }
 
   pickWord() {
     // ... your code goes here
+    return this.secretWord.toString()
   }
 
   checkIfLetter(keyCode) {
     // ... your code goes here
+    return keyCode >= 65 && keyCode <= 90 ? true : false
   }
 
   checkClickedLetters(letter) {
     // ... your code goes here
+    return this.letters.includes(letter) ? false : true
   }
 
   addCorrectLetter(letter) {
     // ... your code goes here
+
+    this.guessedLetters += letter
   }
 
   addWrongLetter(letter) {
     // ... your code goes here
+    this.errorsLeft -= 1
   }
 
   checkGameOver() {
     // ... your code goes here
+    return this.errorsLeft ? false : true
   }
 
   checkWinner() {
     // ... your code goes here
+    function filtraPalabra(palabra) {
+      return palabra
+        .split('')
+        .sort((a, b) => a.localeCompare(b))
+        .filter((item, pos, self) => self.indexOf(item) == pos)
+        .join('');
+    }
+    if (filtraPalabra(this.guessedLetters) === filtraPalabra(this.secretWord)) {
```

Comment:
podes directamente devolver `filtraPalabra(this.guessedLetters) === filtraPalabra(this.secretWord)`